File: Dockerfile.java
FROM maven:3.8-jdk-11 AS build

WORKDIR /code

COPY /java/pom.xml /code/pom.xml
RUN ["mvn", "dependency:resolve"]
RUN ["mvn", "verify"]

# Adding source, compile and package into a fat jar
COPY ["java/src/main", "/code/src/main"]
RUN ["mvn", "package"]

FROM openjdk:11-jre-slim

COPY --from=build /code/target/worker-jar-with-dependencies.jar /

CMD ["java", "-jar", "/worker-jar-with-dependencies.jar"]

File: Dockerfile.go
FROM golang:1.16-alpine as build
WORKDIR /go/src/github.com/votingapp/worker
COPY go/ .
RUN CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -o worker .

FROM scratch
COPY --from=build /go/src/github.com/votingapp/worker/ .
CMD ["./worker"]